tribits/examples/TribitsExampleProject/packages/wrap_external/CMakeLists.txt

```diff
@@ -66,41 +51,51 @@ endif()
 
 string(TOUPPER "${CMAKE_BUILD_TYPE}" upperBuildType)
 
-set(includeDirsList
-  -I ${WithSubpackagesA_SOURCE_DIR}
-  -I ${SimpleCxx_SOURCE_DIR}/src
-  -I ${SimpleCxx_BINARY_DIR}/src
-  )
+get_target_property(simpleCxx_IncludeDirs
+  SimpleCxx::simplecxx  INTERFACE_INCLUDE_DIRECTORIES)
+
+get_target_property(withSubpackagesA_IncludeDirs
+  WithSubpackagesA::pws_a  INTERFACE_INCLUDE_DIRECTORIES)
+
+set(includeDirsList ${withSubpackagesA_IncludeDirs} ${simpleCxx_IncludeDirs})
+
 if (${PACKAGE_NAME}_ENABLE_MixedLang)
-  list(PREPEND includeDirsList
-    -I ${MixedLang_SOURCE_DIR}/src
-    -I ${MixedLang_BINARY_DIR}/src
-    )
+  get_target_property(mixedLang_IncludeDirs
+    MixedLang::mixedlang  INTERFACE_INCLUDE_DIRECTORIES)
+  list(PREPEND includeDirsList ${mixedLang_IncludeDirs})
 endif()
-list(JOIN includeDirsList " " includeDirs)
-# NOTE: TriBITS export Makefile support used to handle the above stuff
-# automatically but that is what you give up when moving to modern CMake.
+list(JOIN includeDirsList " -I" includeDirsCompileOptions)
+set(includeDirsCompileOptions "-I${includeDirsCompileOptions}")
```

Comment:
That does not seem to be a problem.  The tests would break if this code does not work at some point.  When that happens, we can change to generate the final file which will take care of that.